src/watch.ts

```diff
@@ -41,20 +41,46 @@ export class Watch {
         const trxPath = path.join(this.testCommands.testResultFolder, `autoWatch${index}.trx`);
 
         AppInsightsClient.sendEvent("runWatchCommand");
-        const command = `dotnet watch test${Utility.additionalArgumentsOption} --logger "trx;LogFileName=${trxPath}"`;
+        const command = `dotnet watch test ${Utility.additionalArgumentsOption}`
+            + ` --verbosity:quiet` // be less verbose to avoid false positives when parsing output
+            + ` --logger "trx;LogFileName=${trxPath}"`;
 
         Logger.Log(`Executing ${command} in ${testDirectory}`);
         const p = Executor.exec(command, (err: any, stdout: string) => {
             Logger.Log(stdout);
         }, testDirectory, true);
 
-        p.stdout.on("data", (buf) => {
+        let startedLine = [];
+        p.stdout.on("data", async (buf) => {
             const stdout = String(buf);
-            Logger.Log(stdout);
 
-            // Only notify that test are running when a watch has triggered due to changes
-            if (stdout.indexOf("watch : Started") > -1) {
-                this.testCommands.watchRunningTests(namespaceForDirectory);
+            // The string contained in `buf` may contain less or more
+            // than one line. But we want to parse lines as a whole.
+            // Consequently, we have to join them.
+            const lines = [];
+            const lineSegments = stdout.split("\n");
+            for (let i = 0; i < lineSegments.length; i++) {
+                const lineSegment = lineSegments[i];
+                const hadNewline = i > 0;
+                startedLine.push(lineSegment);
+                if (hadNewline) {
+                    const line = startedLine.join("");
+                    startedLine = [];
+                    lines.push(line);
```

Comment:
When I test ran on my machine the lines seems to contain some whitespace which causes the equals check on line 77 to never become true. I guess this might be different on different OS (I'm on windows) but if I add a trim here bofre adding line to lines array it works better. Thoughts from you on this?

Reply:
Hmm, I know that the `dotnet watch` command feeds the strings `"watch : "`, `"Started.\n"` into stdout on my machine. If we were to trim them, we would get something like `"watch :Started"`, which would not match further down.

Could you give me some samples on which strings are being fed into `buf` here on your machine?

Reply:
Oh, sorry, I just realized you meant the classical CRLF line ending issue - line endings are `'\n'` on linux, `'\r'` on Mac and `"\r\n"` on Windows. We should split on both, to avoid running into the same issue on Mac. I'll write some code and push it.

Reply:
Done.